### ISSUE TITLE
Hired pawns in mental state(drug or alcohol breaks)

### ISSUE THREAD

**Issue by artur-shamigulov** [Vanilla-Expanded/VanillaExpandedFramework#5]
Hello, I faced with problem when a contract was ended or cancelled some hired pawn could be in mental states. Usually they get command to leave map, but because it is conflicted with breaks behaviour (drink beer for example), pawns stick in loop with to commands leave map and drink beer(or something else).
else if (pawn.health.capacities.CapableOf(PawnCapacityDefOf.Moving)) {
HiringContractTracker.cs row #73
You could check it there( && !pawn.InMentalState), or simply cancel the state.

**Comment by tixiv** [comment]
Yes, I also ran into this issue. I had two pawns that were mentally broken bug out, they would have died of starvation probably because they were caught in a binge/leave/binge/leave loop and weren't able to finish taking a drug or eating a meal.  And I had one pawn that was technically able to move, but he would take one step out of the hospital bed, then he would need rescue, and so on. Probably only capable of crawling.
I am working on changing the hireable framework to use a quest to give you the pawns. That quest can then nicely track the pawns and manage the leaving of the pawns through standard rimworld ways instead of  the "sudo leave the map, no, don't rest, don't eat, sudo leave the map" approach the framework is using at the moment.
